### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by yelnikov-andrii**

### CODE REVIEW

File: src/herbivoresAndCarnivores.js [hunk 1]
```diff
@@ -1,15 +1,38 @@
...
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100, hidden) {
+    super(name, health, hidden);
```
Comment (Andriyk0) [inline src/herbivoresAndCarnivores.js:14]: you dont need (health) !
you extends this key from Animal!

and hidden write only in constructor!
in super you write key only from extends class

File: src/herbivoresAndCarnivores.js [hunk 2]
```diff
@@ -1,15 +1,38 @@
...
+  hide(hidden) {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+  }
```
Comment (Andriyk0) [inline src/herbivoresAndCarnivores.js:25]: you dont need this

**Review by Andriyk0 — CHANGES_REQUESTED** [review]

File: src/herbivoresAndCarnivores.js [hunk 3]
```diff
@@ -1,15 +1,34 @@
...
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(hidden) {
+    super(hidden);
+    this.hidden = false;
+  }
+  hide(hidden) {
+    this.hidden = true;
+  }
```
Comment (Anna-Tkachenko) [inline src/herbivoresAndCarnivores.js:19]: Herbivore has a method of hide, which changes the hidden property of the beast to the opposite value and helps to hide from carnivores

Comment (Anna-Tkachenko) [reply src/herbivoresAndCarnivores.js:19]: You should change hidden property to the opposite value, not always to the true

**Review by Anna-Tkachenko — CHANGES_REQUESTED** [review]

**Review by Anna-Tkachenko — CHANGES_REQUESTED** [review]
You didn't fix my comments

File: src/herbivoresAndCarnivores.js [hunk 4]
```diff
@@ -1,15 +1,36 @@
...
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(hidden) {
+    super(hidden);
```
Comment (Anna-Tkachenko) [inline src/herbivoresAndCarnivores.js:14]: You shouldn't pass hidden value to the Animal constructor

File: src/herbivoresAndCarnivores.js [hunk 5]
```diff
@@ -1,15 +1,36 @@
...
+  hide(hidden) {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(herb) {
+    if (herb instanceof Herbivore) {
+      if (herb.hidden === false) {
```
Comment (Anna-Tkachenko) [inline src/herbivoresAndCarnivores.js:25]: redundant if
combine two if statements in one main statement using logic operator &&

**Review by Anna-Tkachenko — CHANGES_REQUESTED** [review]

File: src/herbivoresAndCarnivores.js (hunk 4 above)
Comment (Anna-Tkachenko) [reply src/herbivoresAndCarnivores.js:14]: just super()

**Review by Anna-Tkachenko — APPROVED** [review]

**Pull request closed without merging**